form.js

```diff
@@ -0,0 +1,58 @@
+
+/**
+ * Creates a menu entry in the Google Docs UI when the document is opened.
+ * This method is only used by the regular add-on, and is never called by
+ * the mobile add-on version.
+ *
+ * @param {object} e The event parameter for a simple onOpen trigger. To
+ *     determine which authorization mode (ScriptApp.AuthMode) the trigger is
+ *     running in, inspect e.authMode.
+ */
+function onOpen() {
+  DocumentApp.getUi() 
+      .createMenu('Form')
+      .addItem('Open', 'Display')
+      .addToUi();
+}
+
+
+/**
+ * Opens a form in the document containing the add-on's user interface.
+ */
+function Display() {
+  var html = HtmlService.createHtmlOutputFromFile('uploadform')
+      .setWidth(600)
+      .setHeight(425);
+  DocumentApp.getUi() 
+      .showModalDialog(html, 'Upload a File');
+}
+
+
+/**
+ * Gets the blob data, mimetype and filename and decodes the data back 
+ * from base64, then create a blob from all of these and then create a 
+ * file in the drive
+ *
+ * @param {string} data The base64 encoded file data.
+ * @param {string} mimetype The required mimetype of the file.
+ * @param {string} filename The required filename for the file. 
+ * @return {string} string containing the drive url of the file created.
+ */
+function processForm(data, mimetype, filename) {
+  var filedata = Utilities.base64Decode(data);
```

Comment:
are we expecting all data to be base64 encoded?

Reply:
Yes, the file type input in an html form return the file data in base64 decoded format.